lib/index.js

```diff
@@ -2,35 +2,40 @@
 
 const Boom = require('boom');
 const Hoek = require('hoek');
+const Joi = require('joi');
 
 // Declare Internals
 
 const internals = {};
 
-exports.register = (server, options, next) => {
-
-    server.auth.scheme('bearer-access-token', internals.implementation);
-    next();
+internals.defaults = {
+    accessTokenName: 'access_token',
+    allowQueryToken: true,
+    allowCookieToken: false,
+    allowMultipleHeaders: false,
+    allowChaining: false,
+    tokenType: 'Bearer'
 };
 
-exports.register.attributes = {
-    pkg: require('../package.json')
-};
+internals.schema = Joi.object().keys({
+    validateFunc: Joi.func().required(),
+    accessTokenName: Joi.string().required(),
+    allowQueryToken: Joi.boolean(),
+    allowCookieToken: Joi.boolean(),
+    allowMultipleHeaders: Joi.boolean(),
+    allowChaining: Joi.boolean(),
+    tokenType: Joi.string().required()
+});
 
 internals.implementation = (server, options) => {
 
     Hoek.assert(options, 'Missing bearer auth strategy options');
```

Comment:
The real reason to leave this assert, is because now that I am using `Hoek.applyToDefaults` if no options are passed, it will [return null](https://github.com/hapijs/hoek/blob/master/lib/index.js#L156) instead of the default values. So this is just a security guard against that. 

The above problem could be easily solved by doing:

```js
   const settings = Hoek.applyToDefaults(internals.defaults, options || {});
```
However, since I am adding a new code branch, I am not sure how to add a test-case for that branch in the current test file. In that way we could just leverage on the joi's output.